mmocr/datasets/pipelines/textdet_targets/drrg_targets.py

```diff
@@ -0,0 +1,534 @@
+import cv2
+import numpy as np
+from lanms import merge_quadrangle_n9 as la_nms
+from numpy.linalg import norm
+
+import mmocr.utils.check_argument as check_argument
+from mmdet.core import BitmapMasks
+from mmdet.datasets.builder import PIPELINES
+from .textsnake_targets import TextSnakeTargets
+
+
+@PIPELINES.register_module()
+class DRRGTargets(TextSnakeTargets):
+    """Generate the ground truth targets of DRRG: Deep Relational Reasoning
+    Graph Network for Arbitrary Shape Text Detection.
+
+    [https://arxiv.org/abs/2003.07493]. This code was partially adapted from
+    https://github.com/GXYM/DRRG licensed under the MIT license.
+
+    Args:
+        orientation_thr (float): The threshold for distinguishing between
+            head edge and tail edge among the horizontal and vertical edges
+            of a quadrangle.
+        resample_step (float): The step size for resampling the text center
+            line.
+        min_comp_num (int): The minimum number of text components, which
+            should be larger than k_hop1 mentioned in paper.
+        max_comp_num (int): The maximum number of text components.
+        min_width (float): The minimum width of text components.
+        max_width (float): The maximum width of text components.
+        center_region_shrink_ratio (float): The shrink ratio of text center
+            regions.
+        comp_shrink_ratio (float): The shrink ratio of text components.
+        text_comp_ratio (float): The reciprocal of aspect ratio of text
```

Comment:
Sorry what does "reciprocal of aspect ratio" mean? w/h vs h/w?

Reply:
w/h

Reply:
I don't know. Wiki says aspect ratio is w/h
https://en.wikipedia.org/wiki/Aspect_ratio_(image)

My point is, either the `_ratio` or "aspect ratio" is a bad name. Because we cannot guess who divides who.

Reply:
Maybe it should be named `comp_w_h_ratio`?

Reply:
Yeah better